### DISCUSSION TITLE
Walletclient instance support watch chainChange & accoutChange ?

### DISCUSSION THREAD

**Discussion by zouhangwithsweet** [wevm/viem#186] [Idea / Feature Request] [answered]
I think the walletClient can add some event listeners or have an api to watch event change?

**Comment by jxom** [comment, marked as answer]
Could you not just use the provider directly? 

E.g.
```
window.ethereum.addEventListener(…)
```

viem doesn’t use EIP-1193 event listeners, so I don’t think it’s worth implementing.

**Reply by zouhangwithsweet** [reply]
OK, I got it
